AF-308 Initial implementation

robbecker-wf opened the pull request:
# Overview

A CLI that provides dart 1 and dart 2 compatible build and serve commands.

This tool is intended to be used as a transition tool when moving from Dart 1
to Dart 2. Since the `pub serve` and `pub build` commands are not present in
Dart 2, you can use dart_build to run the same command under both versions of
Dart, thus easing the transition to Dart 2. With dart_build you don't need to
modify your pubspec.yaml to add build_runner and build_web_compilers to be
able to build under Dart 2. Those dependencies are included by adding a
dependency on dart_build.

# Changes

  - Copied relevant build_runner pieces from webdev for Dart 2 build and serve
  - Add in dart 1 ability to run pub build & serve
  - Update licensing to include Workiva + original Dart team license

# Testing
 - Clone this repo and checkout this branch
 - Add it as a dev dependency to another project
```
dev_dependencies:
  dart_build:
    path: ../dart_build
```
 - pub get in the other project (with Dart 1)
 - verify that `pub run dart_build serve` runs a server
 - verify that `pub run dart_build build` builds

The dart2 version is on the dart2 branch and updates dependencies for build_runner and SDK constraints to Dart 2. Diff can be viewed here https://github.com/Workiva/dart_build/compare/initial_dart1_functionality...dart2

robbecker-wf committed:
Initial implementation

robbecker-wf committed:
update version in readme

rmconsole-wf commented:
## General Information
Ticket(s): 
* [AF-308](https://jira.atl.workiva.net/browse/AF-308) | [remove](https://w-rmconsole.appspot.com/api/v1/pulls/Workiva/dart_build/1/remove_ticket/AF-308/)

Code Review(s): https://github.com/Workiva/dart_build/pull/1
- [ ] All commits reviewed (322723ea3789b09d57582438a71a9c1cd1dd9818 authored by robbecker-wf)
- [ ] QA Review (322723ea3789b09d57582438a71a9c1cd1dd9818)
- [ ] Dependency Scan Clean ([smithy build](https://ci.webfilings.com/build/1283175) not yet successful)

Reviewers: None

## Additional Information
Watchlist Notifications: None

		When this pull is merged I will add it to the following release:
		Version: dart_build 0.1.0
		Release Ticket(s): None
## 

***
*[Last updated on Tuesday, May 15 03:57 PM CST](https://w-rmconsole.appspot.com/pulls/Workiva/dart_build/1)*

aviary2-wf commented:
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

robbecker-wf committed:
add Dockerfile for Workiva build

robbecker-wf committed:
allow 1.24.2 in SDK constraints

robbecker-wf commented on pubspec.yaml:
```diff
@@ -0,0 +1,29 @@
+#   sdk: ">=2.0.0-dev.32.0 <2.0.0" # TODO update once Dart 2 stable releases
+
+dependencies:
+  args: ^1.2.0
```
Will need to either expand this range to ">=0.13.3 < 2.0.0" or get Workiva up to date to args ^1.0.0

robbecker-wf committed:
widen args range so it works for dart 1 and dart 2

robbecker-wf committed:
add NOTICE and +dart1 version to pubspec.yaml

robbecker-wf committed:
simplify and support all arguments
- remove unused dependencies
- consolidate code into 1 file
- Manually use new ArgParser.allowAnything() instead of command classes to allow any args for either command

robbecker-wf committed:
add git ignore file

robbecker-wf committed:
get rid of args dependency all together

robbecker-wf committed:
fix CI

robbecker-wf committed:
fix CI again

robbecker-wf committed:
fix hung process when pub serve/build outputs an error

robbecker-wf committed:
update pubspec to the specific version that removed pub serve/build

robbecker-wf committed:
handle port argument mismatches between pub and build_runner

robbecker-wf committed:
cleanup
- some comments
- conditional argument adding for build

robbecker-wf committed:
non-interactive build mode by default, fix default port and..
make serve log requests like pub serve does

robbecker-wf committed:
polish things up
- add travis CI file
- abide update & fix lints
- update readme to public pubspec info and more info

robbecker-wf commented:
We aren't going to need this. dart_dev handles most of it and we can move devs to standard dart 2 tooling.
